addon/components/container-query.ts

```diff
@@ -24,7 +24,24 @@ interface ContainerQueryComponentArgs {
   tagName?: string;
 }
 
-export default class ContainerQueryComponent extends Component<ContainerQueryComponentArgs> {
+interface Signature {
+  Element: HTMLElement;
+  Args: ContainerQueryComponentArgs;
+  Blocks: {
+    default: [
+      {
+        features: Features;
+        dimensions: {
+          aspectRatio: number;
+          height: number;
+          width: number;
+        };
+      }
+    ];
+  };
+}
```

Comment:
@NullVoxPopuli Can you rebase your branch so that you can define the signature while assuming the latest code?

I believe we'll want:

```ts
interface ContainerQueryComponentSignature {
  Args: {
    dataAttributePrefix?: string;
    debounce?: number;
    features?: Features;
    tagName?: string;
  };
  Blocks: {
    default: [
      {
        dimensions: Dimensions;
        features: Features;
      }
    ];
  };
  Element: HTMLElement;
}
```

Reply:
**questions**

1\. Is there a naming convention for the signature? In other words, by default, is it always called `Signature`, or would it be `ContainerQueryComponentSignature` (should the component be generated using Ember CLI now, or in the future)?

2\. Can you clarify for me what you had meant by `@glimmer/component@v1.1.2`? It is okay for the addon to use `ContainerQueryComponentSignature` now? Or is the signature valid only when the consuming app (or addon) has `@glimmer/component@v1.1.2` installed?

Reply:
1. not sure. personally, I'd prefer just _Signature_, as then I don't need to name each and every one. But I also don't know what is planned for the generator. @chriskrycho would know tho

2. The signature style of components _requires_ at least glimmer/component 1.1.2. Prior to that, the signature type will error in typescript projects using prior to 1.1.2

Reply:
I included this change in a recent patchfile I made so I could continue upgrading https://limber.glimdown.com https://github.com/NullVoxPopuli/limber/pull/621/files#diff-b2a12a590e7e29c8f846dbc02ddc7973ccd718121f8bda6e44ca9658aeb2a650

(I'll be updating _usage_ of my patched ember-container-query in a future PR)

:tada: